ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/PendingPeerRequest.java

```diff
@@ -57,6 +57,9 @@ public boolean attemptExecution() {
       // No peers have the required height.
       result.completeExceptionally(new NoAvailablePeersException());
       return true;
+    } else if (maybePeer.get().isDisconnected()) {
+      result.completeExceptionally(new PeerDisconnectedException(maybePeer.get()));
+      return true;
```

Comment:
I don't understand why this is needed. Won't the PeerNotConnected exception in the sendRequest end up throwing this is the peer is disconnected? Seems like this is doubling up on the disconnected check.

Reply:
Yeah true, although this check is done before we check if the peer has capacity. It probably matches previous functionality better if we remove this check though. What do you think?

Reply:
Given the complexity and risk of changes of breaking things in this area of code, I think it does make more to do this later after the refactoring and also probably after changing the retrying tasks to switching retrying tasks https://github.com/hyperledger/besu/issues/5271